kernel/common/method.rb

```diff
@@ -158,6 +159,23 @@ def unbind
     UnboundMethod.new(@defined_in, @executable, @receiver.class, @name)
   end
 
+  def super_method
+    @ancestors.each_with_index do |a, i|
+      if a.methods.include?(@name)
+        method = a.method(@name).bind(a)
+      elsif a.instance_methods.include?(@name)
+        method = a.instance_method(@name).bind(@receiver)
+      end
+
+      if defined?(method)
+        method.instance_variable_set(:@ancestors, @ancestors.drop(i+1))
```

Comment:
OK if stuff like this is required we'll _definitely_ need to do some extra plumbing to support this nicely. Injecting instance variables into another object is a no-no. Thinking of it this also poses a synchronization problem: what if two threads end up at this code at the same time?